sweep/weight_estimator.go

```diff
@@ -38,3 +95,18 @@ func (w *weightEstimator) addP2WKHOutput() {
 func (w *weightEstimator) weight() int {
 	return w.estimator.Weight()
 }
+
+// fe returns the tx fee to use for the aggregated inputs and outputs, taking
+// into account unconfirmed parent transactions (cpfp).
+func (w *weightEstimator) fee() btcutil.Amount {
+	// Calculate fee and weight for just this tx.
+	weight := int64(w.estimator.Weight())
+
+	// Add combined weight of unconfirmed parent txes.
+	weight += w.parentsWeight
+
+	// Subtract fee already paid by parents.
+	fee := w.feeRate.FeeForWeight(weight) - w.parentsFee
```

Comment:
it seems possible that this could be negative? or is the theory that this should never do that because of the `parentFeeRate >= w.feeRate` check? i wonder if that is always true when considering rounding edge cases

Reply:
I don't think there is a rounding edge case, because even in the case that the parent almost totally pays for itself, the child tx has a minimum weight too that needs to be covered.

Reply:
An alternative would be to remove the `parentFeeRate >= w.feeRate` check, but here instead return the maximum of `feeRate.FeeForWeight(weight)` and `feeRate.FeeForWeight(weight+parentsWeight)-parentsFee`

Reply:
> even in the case that the parent almost totally pays for itself, the child tx has a minimum weight too that needs to be covered.

the minimum weight of a transaction is less than 1000, so it still seems possible to me. would be nice to know for sure tho

Reply:
>An alternative would be to remove the parentFeeRate >= w.feeRate check, but here instead return the maximum of feeRate.FeeForWeight(weight) and feeRate.FeeForWeight(weight+parentsWeight)-parentsFee

I don't think that is possible, because there may be multiple inputs with different parents, some of which can be cpfp'ed and others not.

Reply:
No, you are right. The floor should be what is required for just the child tx. Added that.